File: old_groovy/src/main/groovy/net/zomis/brainf/model/ast/tree/SteppableSyntax.java
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner;

public interface SteppableSyntax {

    int getValue();
    void performTimes(BrainfuckRunner runner, int steps);
    default int getTimes() {
        return Math.abs(getValue());
    }

}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/tree/SteppableSyntax.kt
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner;
import kotlin.math.absoluteValue

interface SteppableSyntax {

    fun getValue(): Int
    fun performTimes(runner: BrainfuckRunner, steps: Int)
    val times get() = getValue().absoluteValue

}
